test/suite/functional/Output/Copy/Serializer/BodySerializerTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoclerLabs\ApiClientGenerator\Test\Functional\Output\Copy\Serializer;
+
+use DoclerLabs\ApiClientException\UnexpectedResponseException;
+use DoclerLabs\ApiClientGenerator\Output\Copy\Serializer\BodySerializer;
+use DoclerLabs\ApiClientGenerator\Output\Copy\Serializer\ContentType\FormUrlencodedContentTypeSerializer;
+use DoclerLabs\ApiClientGenerator\Output\Copy\Serializer\ContentType\Json\Json;
+use DoclerLabs\ApiClientGenerator\Output\Copy\Serializer\ContentType\JsonContentTypeSerializer;
+use GuzzleHttp\Psr7\Response;
+use GuzzleHttp\Psr7\Stream;
+use PHPUnit\Framework\TestCase;
+
+class BodySerializerTest extends TestCase
```

Comment:
IMO this looks more like a UT, so maybe move it to test/suite/unit 🤔

Reply:
This is a functional test for me, because it does not use any mocks, and this is intentional. Things are much more clear, and the test is more useful overall in this case.